hs_core/hydroshare/resource.py

```diff
@@ -354,11 +359,13 @@ def create_resource(
 
         fed_zone_home_path = ''
         if fed_res_path:
-            fed_zone_home_path = fed_res_path
             resource.resource_federation_path = fed_res_path
+            fed_zone_home_path = fed_res_path
             resource.save()
-        elif fed_res_file_names:
-            fed_zone_home_path = utils.get_federated_zone_home_path(fed_res_file_names[0])
+
+        # TODO: this option should be eliminated. Require explicit zone path
+        elif len(source_names) > 0:
```

Comment:
@alvacouch On your TODO comment, I believe this option cannot be eliminated since the workflow is to allow users to browse into any iRODS zone, then we detect whether it comes from a federated zone, and if it is, then we get the federated zone path based on ```source_names``` which are the iRODS absolute logic path to selected iRODS files, then pass source_names to create_resource() function which is used to form the resource_federation_path.